common/src/main/java/io/netty/util/ResourceLeakDetector.java

```diff
@@ -91,6 +89,8 @@ static Level parseLevel(String levelStr) {
 
     private static Level level;
 
+    private static int targetRecords;
```

Comment:
is this intentionally not volatile? we may have visibility issues here, and the volatile read may increase cost of leak detection.

Reply:
Being able to change the leak level dynamically seems like an expensive ability for not much payoff.   I would feel more comfortable if I knew the expected use case.

Reply:
@kiril-me can you give some details ?